wavs-staking-template/dev_notes.md

```diff
@@ -0,0 +1,197 @@
+```
+
+#### WAVS-middleware
+
+```bash
+git clone --depth 1 git@github.com:Lay3rLabs/wavs-middleware.git
+cd wavs-middleware
+docker build . -t wavs-middleware:local 
+```
+
+## Chain Setup
+
+### Local Fork
+
+For testnet fork, we need to run anvil and create a private key. All operators must be on the same machine and point to this URL
+
+```bash 
+anvil --fork-url https://ethereum-holesky-rpc.publicnode.com
+```
+
+(You can also use another rpc endpoint of your choice )
+
+We need to create a new private key and fund it for later use. 
+If this is really testnet/mainnet, you can use a more secure key, for testing, just make a one-off key
+
+```bash
+export PRIVATE_KEY=$(cast wallet new --json | jq -r .[0].private_key)
+export MY_ADDR=$(cast wallet address --private-key $PRIVATE_KEY)
+cast rpc anvil_setBalance $MY_ADDR 0x10000000000000000000 -r http://localhost:8545
+```
+
+### Real Testnet
+
+You must create an account and fund it with testnet ETH. You can use MetaMask or just the commands above.
+Getting testnet ETH is out of scope of this guide (TODO: Ron - where to get it)
+
+You will also want to set a testnet rpc url to use it (TODO: link to some good ones, which env var)
+
+## Deploying as AVS Dev Team
+
+More detailed instructions on commands and setup can be found in [wavs-middleware](https://github.com/Lay3rLabs/wavs-middleware/blob/dev/docker/README.md).
+
+You need to prepare an .env file in this directory for running the rest of the commands.
+
+```bash
+cp env.example .env
+# edit the file and make any changes you need there
+```
+
+### Deploy eigen middleware
+
+```bash
+docker run --rm --network host --env-file .env  -v ./.nodes:/root/.nodes wavs-middleware:local
+export SERVICE_MANAGER_ADDRESS=$(jq -r .addresses.WavsServiceManager .nodes/avs_deploy.json)
+```
+
+### Build Custom Application
+
+Assuming you are using the template app, you can do the following. More advanced apps will have custom build steps, run from the top-level directory:
+
+```bash
+cd ..
+make setup
+forge build
+make wasi-build
+```
+
+TODO: `make wasi-build` now fails for me with `(jco componentize) ComponentError: package 'wasi:io@0.2.0' not found. no known packages.`
+
+### Deploy ServiceHandler
+
+```bash
+cd src/contracts
+forge create SimpleSubmit --broadcast --rpc-url http://127.0.0.1:8545 --private-key "$PRIVATE_KEY" --constructor-args "$SERVICE_MANAGER_ADDRESS"
+
+forge create SimpleTrigger --broadcast --rpc-url http://127.0.0.1:8545 --private-key "$PRIVATE_KEY"
+```
+
+Store "deployed_to" from the first command as `SERVICE_HANDLER_ADDRESS`.
+And store "deployed_to" from the second command as `TRIGGER_ADDRESS`.
+
+### Start aggregator
+
+In another window, Go to `aggregator` dir in `wavs-foundry-template` (you may want to edit `aggregator/.wavs_env` first)
+
+```bash
+cd wavs-staking-template/aggregator
+docker compose up
+```
+
+TODO: the following is no longer valid - we need to add the servie.json later
```

Comment:
the flow is:

1. Upload the service JSON somewhere (you can use WAVS itself for convenience, at the `save-service` endpoint)
2. Set the URL to this JSON on the service manager contract
3. Call `add-service` with the `ChainName` (from config toml) and `ServiceManager` address

The new `add-service` request type is here: https://github.com/Lay3rLabs/WAVS/blob/ee5b1107148284dbd56c843ac3522a0ad2e58572/packages/types/src/http.rs#L19

`save-service` is just a post request with the service itself 

This is all seen in the CLI client which combines these for the "deploy service raw" command here: https://github.com/Lay3rLabs/WAVS/blob/ee5b1107148284dbd56c843ac3522a0ad2e58572/packages/cli/src/clients.rs#L79